test/unit_tests/braket/circuits/test_circuit.py

```diff
@@ -2170,6 +2170,21 @@ def test_circuit_with_expr_not_fully_bound():
     assert new_circ == expected
 
 
+def test_to_openqasm_cached():
+    circ = Circuit().h(0).cnot(0, 1)
+    circ.to_ir(IRType.OPENQASM)
+    fake = "foo"
+    circ._cached_openqasm = fake
+    assert circ.to_ir(IRType.OPENQASM) == fake
+
+
+def test_to_openqasm_recompute():
+    circ = Circuit().h(0).cnot(0, 1)
+    circ.to_ir(IRType.OPENQASM)
+    circ._cached_openqasm = "foo"
+    assert circ.to_ir(IRType.OPENQASM, recompute=True) == Circuit(circ).to_ir(IRType.OPENQASM)
```

Comment:
maybe this test would be more robust if we asserted against a string. this seems like it could break ie if the recompute flag was ignored (foo == foo)